incubator/hnc/pkg/validators/hierarchical_namespace.go

```diff
@@ -0,0 +1,119 @@
+package validators
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"k8s.io/api/admission/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	api "github.com/kubernetes-sigs/multi-tenancy/incubator/hnc/api/v1alpha1"
+	"github.com/kubernetes-sigs/multi-tenancy/incubator/hnc/pkg/forest"
+	"github.com/kubernetes-sigs/multi-tenancy/incubator/hnc/pkg/reconcilers"
+)
+
+// HierarchicalNamespaceServingPath is where the validator will run. Must be kept in sync with the
+// kubebuilder markers below.
+const (
+	HierarchicalNamespaceServingPath = "/validate-hnc-x-k8s-io-v1alpha1-hierarchicalnamespaces"
+)
+
+// Note: the validating webhook FAILS CLOSE. This means that if the webhook goes down, all further
+// changes are forbidden.
+//
+// +kubebuilder:webhook:path=/validate-hnc-x-k8s-io-v1alpha1-hierarchicalnamespaces,mutating=false,failurePolicy=fail,groups="hnc.x-k8s.io",resources=hierarchicalnamespaces,verbs=create;update;delete,versions=v1alpha1,name=hierarchicalnamespaces.hnc.x-k8s.io
```

Comment:
why the `update` is needed? In your `Handle` method, it only handles `create` and `delete` operations.

Reply:
Nice catch! We should probably reject any updates to HNSes (it doesn't have any real fields anyway), or else take `update` out of this list.

Reply:
Good catch, thanks! Removed.

Reply:
I manually tested with `update` removed. Seems to reject all `update` requests, we need to have it listed here and in `Handle` reject all `update` requests.

Reply:
How did you try to update it? You mean with labels and stuff? Yeah, perhaps we should allow those.

I'm surprised that an action that we _didn't_ list as part of the validator would be blocked. What was the error message with `update` removed from the webhook config?

Reply:
Note that `update` is still present here.

Reply:
> How did you try to update it? You mean with labels and stuff? Yeah, perhaps we should allow those.

I used `kubectl edit hns child -n parent`
> I'm surprised that an action that we _didn't_ list as part of the validator would be blocked. What was the error message with `update` removed from the webhook config?

The action not listed will be `allowed`, not blocked. When `update` removed, I could successfully edit the hns instance.

Reply:
Ah, I see. I think we should allow them to let labels and annotations be added, wdyt?

Reply:
But that's by HNC SA, right? Not by users? We only deny `update` if it's not from SA.

Reply:
No, if a user wants to go add a couple of labels to the HNS objects (_not_ the namespaces), I don't see why we'd stop them.

Reply:
Got you. Sure, then I just need to remove the `update` verb and clean it up in `handle`.